src/nvim/fileio.c

```diff
@@ -1172,28 +1173,20 @@ int readfile(char *fname, char *sfname, linenr_T from, linenr_T lines_to_skip,
 
 #ifdef HAVE_ICONV
       if (iconv_fd != (iconv_t)-1) {
-        /*
-         * Attempt conversion of the read bytes to 'encoding' using
-         * iconv().
-         */
-        const char *fromp;
-        char *top;
-        size_t from_size;
-        size_t to_size;
-
-        fromp = ptr;
-        from_size = size;
+        // Attempt conversion of the read bytes to 'encoding' using iconv().
+        const char *fromp = ptr;
+        long from_size = size;
```

Comment:
I've heard the same thing about `int`, `size_t` and basically all facets of C programming language. Your suggestion isn't compelling. *Why* is `long` worse? If it is bad, shouldn't it be removed from the codebase in general?

I used `long` because `size` is of type `long` and because unsigned arithmetic is "bad".